docs/topics/gradle/gradle-configure.md

```diff
@@ -0,0 +1,745 @@
+
+### Kotlin and Java sources
+
+Kotlin sources and Java sources can be stored in the same folder, or they can be placed in different folders. The default convention is to use different folders:
+
+```groovy
+project
+    - src
+        - main (root)
+            - kotlin
+            - java
+```
+
+The corresponding `sourceSets` property should be updated if you are not using the default convention:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+sourceSets.main {
+    java.srcDirs("src/main/myJava", "src/main/myKotlin")
+}
+```
+
+</tab>
+<tab title="Groovy" group-key="groovy">
+
+```groovy
+sourceSets {
+    main.kotlin.srcDirs += 'src/main/myKotlin'
+    main.java.srcDirs += 'src/main/myJava'
+}
+```
+
+</tab>
+</tabs>
+
+### Check for JVM target compatibility of related compile tasks
+
+In the build module, you may have related compile tasks, for example:
+* `compileKotlin` and `compileJava`
+* `compileTestKotlin` and `compileTestJava`
+
+> `main` and `test` source set compile tasks are not related.
+>
+{type="note"}
+
+For such related tasks, the Kotlin Gradle plugin checks for JVM target compatibility. Different values of `jvmTarget` in
+the `kotlin` extension and [`targetCompatibility`](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)
+in the `java` extension cause incompatibility. For example:
+the `compileKotlin` task has `jvmTarget=1.8`, and
+the `compileJava` task has (or [inherits](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)) `targetCompatibility=15`.
+
+Control the behavior of this check by setting the `kotlin.jvm.target.validation.mode` property in the `build.gradle`
+file equal to:
+
+* `warning` – the default value; the Kotlin Gradle plugin will print a warning message.
+* `error` – the plugin will fail the build.
+* `ignore` – the plugin will skip the check and won't produce any messages.
+
+### Set custom JDK home
+
+By default, Kotlin compile tasks use the current Gradle JDK.
+If you need to change the JDK by some reason, you can set the JDK home with [Java toolchains](#gradle-java-toolchains-support)
+or the [Task DSL](#setting-jdk-version-with-the-task-dsl) to set a local JDK.
+
+> The `jdkHome` compiler option is deprecated since Kotlin 1.5.30.
+>
+{type="warning"}
+
+When you use a custom JDK, note that [kapt task workers](kapt.md#running-kapt-tasks-in-parallel)
+use [process isolation mode](https://docs.gradle.org/current/userguide/worker_api.html#changing_the_isolation_mode) only,
+and ignore the `kapt.workers.isolation` property.
+
+### Gradle Java toolchains support
+
+Gradle 6.7 introduced [Java toolchains support](https://docs.gradle.org/current/userguide/toolchains.html).
+Using this feature, you can:
+* Use a JDK and a JRE that are different from the Gradle ones to run compilations, tests, and executables.
```

Comment:
```suggestion
* Use a JDK and a JRE that are different from the Gradle versions to run compilations, tests, and executables.
```

Reply:
These "versions" are not the versions of Gradle itself, but Gradle has its own JDK and JRE, and about these versions we're talking about.

Reply:
Ah OK! Updated the suggestion based on your feedback.

Reply:
Thank you! But there is still such a point: people usually don't have any idea which JDK and JRE Gradle uses because Gradle does it automatically. You don't plug JDK into Gradle, it goes packed with a Gradle version. Why can't we leave the sentence as it was?

Reply:
I suggested the change because "the Gradle ones" isn't correct. A definite article is needed. The closest substitute I can think of is "the ones in Gradle". Maybe this article about "ones" will help explain what I mean: https://langeek.co/en/grammar/course/175/one-and-ones

Reply:
the Gradle's ones?

I used "the ones in Gradle".

Reply:
Ah yes. This comment snippet is out of date. This looks great!